noxfile.py

```diff
@@ -289,6 +289,10 @@ def _get_pinned_requirements(
         if req.startswith("numpy") and _numpy is not None:
             print("adding numpy constraint <2")
             req = f"{req}, {_numpy}\n"
+        if req == "pyarrow" or req.startswith("pyarrow "):
+            req = "pyarrow >= 13\n"
+        if req == "ibis-framework" or req.startswith("ibis-framework "):
+            req = "ibis-framework[duckdb]\n"
```

Comment:
@cosmicBboy I think you're already working on this, based on your comment on a previous PR, but this feels extremely complicated and confusing. 😅 I think this is a reasonable set of overrides for now, and the actual package dependencies look correct, but your validation on this especially would be great. :)